### ISSUE TITLE
WriteInt16_StructWithReferenceTypes_ReturnsExpected test failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#38353]
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38347~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190607.42/workItem/System.Runtime.InteropServices.Tests/analysis/xunit/System.Runtime.InteropServices.Tests.Int16Tests~2FWriteInt16_StructWithReferenceTypes_ReturnsExpected
```
Windows.81.Amd64.Open-x86-Release
Get Repro environment
Unhandled Exception of Type Xunit.Sdk.NotEqualException
Message :
Assert.NotEqual() Failure
Expected: Not 0
Actual:   0
Stack Trace :
   at System.Runtime.InteropServices.Tests.Int16Tests.WriteInt16_StructWithReferenceTypes_ReturnsExpected() in /_/src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ReadWrite/Int16Tests.cs:line 88
```
cc: @AaronRobinsonMSFT

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky Opinions on this area? This must be a flaky test issue. I am unsure why this would be the issue though.

**Comment by jkoritzinsky** [comment]
Looks like the same issue as #34422. We should either remove the assert that failed since we aren't really testing that offset with the `WriteInt16` call or do the same thing we did to fix #34422.

**Comment by AaronRobinsonMSFT** [comment]
Sounds good. Let's just remove it.
